helix-core/src/test/java/org/apache/helix/integration/task/TestWorkflowTermination.java

```diff
@@ -62,7 +62,7 @@ public void beforeClass() throws Exception {
     String taskState = shouldJobFail ? TaskState.FAILED.name() : TaskState.COMPLETED.name();
     return new JobConfig.Builder().setTargetResource(WorkflowGenerator.DEFAULT_TGT_DB)
         .setTargetPartitionStates(Sets.newHashSet(MasterSlaveSMD.States.MASTER.name()))
-        .setWorkflow(workflow).setCommand(MockTask.TASK_COMMAND)
+        .setWorkflow(workflow).setCommand(MockTask.TASK_COMMAND).setMaxAttemptsPerTask(5)
```

Comment:
Why we want to limit this?

Reply:
testWorkflowJobFail is polling for the Failed state for 10 seconds. However, I noticed the tasks in job3 of this workflow need to be retried for 10 times and then this job will go to Failed state. However, 10 seconds is little bit short for this case (for other jobs 1 and 2 and possibly 4 to run and job 3 to be retried 10 times and then failed). I noticed that the job fails right after 10 seconds. Hence, in this PR the maxNumberOfAttempts has reduced set to 5 for the tasks.